manager/controlapi/service.go

```diff
@@ -819,6 +856,13 @@ func (s *Server) UpdateService(ctx context.Context, request *api.UpdateServiceRe
 
 		service.Meta.Version = *request.ServiceVersion
 
+		// if the service has a JobStatus, that means it must be a Job, and we
+		// should increment the JobIteration
```

Comment:
Could be made clearer by just defining `isJob := isJobSpec(service.Spec)` on line 720 above and then the line below just becomes `if isJob`, which is self explanatory and doesn't need this comment any more :) ?